### PR TITLE
Extract edit distance methods to their own helper class

### PR DESCRIPTION
**Pull request by TheRealRavager**
Edit distance calculation is heavily coupled with Fuzzy Find. This PR extracts them out. To be merged after #174

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -317,56 +317,53 @@ Edits the phone number of the specified patient to be `91234567`.
...
-
-__Use case demonstrating keyword case insensitivity__ +
-* `find n/aLiCe` +
-Displays the same result as the preceding example.
+Lets take a look at some examples and pointers showcasing the use of `find`:
 
 __Use case with multiple keywords per prefix__ +
-* `find n/Alice n/Benson b/A b/B b/O` +
-Displays all persons whose name contains either 'Alice' or 'Benson' and whose blood type is either 'A', 'B' or 'O'.
+* `find n/Laura n/Marisha Ray b/A b/B b/O` +
```
Comment (C-likethis123) [inline docs/UserGuide.adoc]: After my PR for signed blood types is merged, you will need to update the user guide to reflect this.

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -317,56 +317,53 @@ Edits the phone number of the specified patient to be `91234567`.
...
+Displays all persons whose name contains either 'Laura' or 'Marisha Ray' and whose blood type is either 'A', 'B' or 'O'.
 
 __Use case demonstrating prefix order insensitivity__ +
-* `find b/A n/Benson b/O n/Alice b/B` +
+* `find b/A n/Benson Carter b/O n/Alice b/B` +
```
Comment (C-likethis123) [inline docs/UserGuide.adoc]: As per comment on signed blood types

File: src/main/java/organice/commons/util/LevenshteinDistanceUtil.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+     * Returns the minimum Levenshtein Distance of every {@code stringFromList} and {@code otherString} pair. If
+     * {@code stringFromList} is a single-word string, {@code otherString} is split according to spaces (if possible)
+     * and the minimum of the minimum distances of every possible pair is returned.
+     */
+    public static int findMinLevenshteinDistance(List<String> stringList, String otherString) {
+        // If keyword is one word long, split the personName and find minLD.
+        // Else just do as we are normally doing in original finalMinLD
+        BiFunction<String, String, Integer> findDistanceSplitIfMultiWord = (stringFromList, oString) ->
+                stringFromList.split(" ").length == 1 ? Arrays.stream(oString.split(" "))
+                        .reduce(Integer.MAX_VALUE, (minDistance, nextWord) -> Integer.min(minDistance,
```
Comment (C-likethis123) [inline src/main/java/organice/commons/util/LevenshteinDistanceUtil.java]: Can I confirm that the algorithm works this way for minLD:
If there are two strings 'Hans Johnson' and 'Hans Jane', it splits the two strings into two string arrays ['Hans', 'Johnson'] and ['Hans', 'Jane']. Then calculates the minLD for: ('Hans', 'Hans'), ('Hans', 'Jane') and ('Johnson', 'Hans'), ('Johnson', 'Jane')?

File: src/main/java/organice/logic/parser/AddressBookParser.java [hunk 4]
```diff
@@ -58,6 +58,9 @@ public Command parseCommand(String userInput) throws ParseException {
         case ClearCommand.COMMAND_WORD:
             return new ClearCommand();
 
+        case FuzzyFindCommand.COMMAND_WORD:
+            return new FuzzyFindCommandParser().parse(arguments);
+
         case FindCommand.COMMAND_WORD:
```
Comment (C-likethis123) [inline src/main/java/organice/logic/parser/AddressBookParser.java]: Just wondering if you would like to rename 'FindCommand' to 'ExactFindCommand' for clarity.

File: src/main/java/organice/logic/parser/FuzzyFindCommandParser.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FuzzyFindCommand parse(String args) throws ParseException {
+        args = args.replaceAll("\n", " ").replaceAll("\\s+", " ");
+        String trimmedArgs = args.trim();
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_NRIC, PREFIX_PHONE,
+                PREFIX_TYPE, PREFIX_AGE, PREFIX_PRIORITY, PREFIX_BLOOD_TYPE, PREFIX_DOCTOR_IN_CHARGE,
+                PREFIX_TISSUE_TYPE, PREFIX_ORGAN_EXPIRY_DATE, PREFIX_ORGAN);
+
+        // Preamble length > 7 indicates initial command is not `fuzfind PREFIX/KEYWORD`
```
Comment (C-likethis123) [inline src/main/java/organice/logic/parser/FuzzyFindCommandParser.java]: May I ask, what is the purpose of this line of code? If I'm not wrong, you could declare a String constant `COMMAND_WORD` in the `FuzzyFindCommand` class and use that.

**Comment by TheRealRavager** [comment]
Lack of time. Closing

**Pull request closed without merging**